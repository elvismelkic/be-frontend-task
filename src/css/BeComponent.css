.be-component {
  width: 10%;
  height: 50px;
  margin-bottom: 20px;
  cursor: default;
}
