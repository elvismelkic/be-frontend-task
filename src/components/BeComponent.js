import React, { Component, Fragment } from "react";
import "../css/BeComponent.css";
import { fetchColors } from "../utils/api";

export default class BeComponent extends Component {
  state = {
    loading: true,
    error: null,
    colors: [],
    activeColor: "#000"
  };

  getColors = async () => {
    this.setState({ loading: true });
    try {
      const color1 = await fetchColors();
      const color2 = await fetchColors();

      this.setState({
        colors: [`#${color1.new_color}`, `#${color2.new_color}`],
        loading: false
      });
    } catch (error) {
      console.log(error);
      this.setState({
        error: "There was a problem with your request. Please, try again."
      });
    }
  };

  handleClick = () => {
    if (this.state.activeColor !== "#000") {
      this.setState({ activeColor: "#000" });
    } else {
      let randomColor =
        Math.random() > 0.5 ? this.state.colors[0] : this.state.colors[1];

      this.setState({ activeColor: randomColor });
    }
  };

  componentDidMount() {
    this.getColors();
  }

  render() {
    const { loading, error, activeColor } = this.state;
    return (
      <Fragment>
        {error !== null ? (
          <h1>{error}</h1>
        ) : loading === false ? (
          <div
            className="be-component"
            style={{ backgroundColor: activeColor }}
            onClick={this.handleClick}
          >
            {this.props.text}
          </div>
        ) : (
          <h1>LOADING</h1>
        )}
      </Fragment>
    );
  }
}
