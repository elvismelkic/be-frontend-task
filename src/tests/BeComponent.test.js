import React from "react";
import BeComponent from "../components/BeComponent";
import renderer from "react-test-renderer";

it("changes background color when clicked", () => {
  const component = renderer.create(<BeComponent text={""} />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger callback
  tree.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger callback
  tree.props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
