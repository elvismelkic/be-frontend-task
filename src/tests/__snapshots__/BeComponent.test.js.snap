// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes background color when clicked 1`] = `
<h1>
  LOADING
</h1>
`;
